src/config/Config.zig

```diff
@@ -5294,6 +5319,240 @@ pub const QuickTerminalScreen = enum {
     @"macos-menu-bar",
 };
 
+/// See quick-terminal-size
+pub const QuickTerminalSize = struct {
+    const Self = @This();
+
+    pub const Size = union(UnitKey) {
+        // Absolute value in pixel.
+        pixel: u16,
+
+        // Percentage value relative to the screen size. Allowed value [0-100].
+        percent: u8,
+
+        // Sync with `ghostty_config_quick_terminal_unit_e`.
+        pub const UnitKey = enum(c_int) {
+            pixel,
+            percent,
+        };
+
+        pub const C = extern struct {
+            value: u16,
+            tag: UnitKey,
+        };
+
+        pub fn cval(self: Size) Size.C {
+            return .{
+                .tag = @as(UnitKey, self),
+                .value = res: {
+                    switch (self) {
+                        .percent => |v| {
+                            break :res v;
+                        },
+                        .pixel => |v| {
+                            break :res @intCast(v);
+                        },
+                    }
+                },
+            };
+        }
+
+        fn parseValue(input: []const u8) !Size {
```

Comment:
[Nitpick] Parsing could be improve, `quick-terminal-size: 20%, 30%` will probably fail due to the space after the comma. I could/should fix it, but other config suffers from the same issue and should probably be fixed in a subsequent PR.